### PR TITLE
feat: add administrators task to promote users

### PR DESCRIPTION
**Pull request by ryanfang5**

### CODE REVIEW

File: tests/administrators/test_tasks.py [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    static_time,
+    tmpdir,
+    fake2,
+):
+
+    await fake2.users.create()
+
+    user = await fake2.users.create()
+
+    async with mongo.create_session() as session:
```
Comment (igboyes) [inline tests/administrators/test_tasks.py:23]: Don't need a session for this.

They are slower and data consistency doesn't matter so much for tests.

File: virtool/administrators/data.py [hunk 2]
```diff
@@ -177,3 +177,12 @@ async def delete(self, user_id: str):
             return
 
         raise ResourceNotFoundError()
+
+    async def promote_administrators(self):
+        async with self._db.create_session() as session:
```
Comment (igboyes) [inline virtool/administrators/data.py:182]: Session doesn't add anything here. You are not modifying MongoDB, which is where as transaction is important.

File: virtool/administrators/data.py [hunk 3]
```diff
@@ -177,3 +177,12 @@ async def delete(self, user_id: str):
...
+            async for user in self._db.users.find(
+                {"administrator": True}, session=session
+            ):
```
Comment (igboyes) [inline virtool/administrators/data.py:185]: You can use `users.distinct("_id", {"administrator": True})` here to be more efficient.

File: tests/administrators/test_tasks.py [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+
+    task = PromoteAdministratorsTask(1, data_layer, {}, get_temp_dir())
+
+    await task.run()
+
+    row = await get_row_by_id(pg, Task, 1)
+    assert row.complete is True
+
+    assert (
+        await data_layer.administrators._authorization_client.list_administrators()
```
Comment (igboyes) [inline tests/administrators/test_tasks.py:53]: I think it's better to use `data_layer.administrators.find` here. That way we're testing the same interface that tasks is touching.

**Review by igboyes — CHANGES_REQUESTED** [review]

File: virtool/startup.py [hunk 5]
```diff
@@ -405,4 +407,5 @@ async def startup_tasks(app: Application):
     await tasks_data.create(CleanReferencesTask)
     await tasks_data.create(TimeoutJobsTask)
 
+    await tasks_data.create_periodically(PromoteAdministratorsTask, 600)
```
Comment (ReeceHoffmann) [inline virtool/startup.py:410]: No need to add this here, this method of spawning tasks will be deprecated and removed in a later version.

**Review by igboyes — DISMISSED** [review]

**Review by igboyes — APPROVED** [review]

**Pull request merged**